### PR TITLE
hot fix short dtype bug in primitive

### PR DESCRIPTION
**Pull request by BBuf**
解决特定nvcc版本下编译错误。

/mnt/DATA/csh/oneflow-cuda-bare-metal/oneflow/core/ep/common/primitive/binary_functor.h(276): error: calling a constexpr __host__ function("std::log<short> ") from a __global__ function("oneflow::cuda::elementwise::ApplyGeneric<(int)1,  ::oneflow::ep::primitive::broadcast_elementwise_binary::_NV_ANON_NAMESPACE::BinaryLhsScalarPtrFunctorFactory<( ::oneflow::ep::primitive::BinaryOp)12, short, short> , short, short > ") is not allowed. The experimental flag '--expt-relaxed-constexpr' can be used to allow this.

/mnt/DATA/csh/oneflow-cuda-bare-metal/oneflow/core/ep/common/primitive/binary_functor.h(276): error: identifier "std::log<short> " is undefined in device code

### CODE REVIEW

**Review by jackalcooper — APPROVED** [review]

File: oneflow/core/functional/impl/comm_functor.cpp [hunk 1]
```diff
@@ -210,9 +210,7 @@ class CommBroadcastFunctor {
   Maybe<Tensor> operator()(const std::shared_ptr<one::Tensor>& x, int64_t src_rank,
                            bool inplace) const {
     const auto& rank_group = JUST(RankGroupScope::CurrentRankGroup());
-    std::string device_type_str = JUST(x->device())->type();
-    CHECK_OR_RETURN(device_type_str == "cuda" || device_type_str == "cpu");
-    DeviceType device_type = device_type_str == "cuda" ? DeviceType::kCUDA : DeviceType::kCPU;
+    DeviceType device_type = JUST(x->device())->enum_type();
```
Comment (marigoold) [inline oneflow/core/functional/impl/comm_functor.cpp:213]: 这里是不是把别的 PR 混进来了

File: oneflow/core/ep/cpu/primitive/fill.cpp [hunk 2]
```diff
@@ -66,7 +66,7 @@ class FillFactoryImpl : public FillFactory {
 #define MAKE_NEW_FILL_ENTRY(type_cpp, type_proto) {type_proto, NewFill<type_cpp>},
 
     static const std::map<DataType, std::function<std::unique_ptr<Fill>()>> new_fill_handle{
-        OF_PP_FOR_EACH_TUPLE(MAKE_NEW_FILL_ENTRY, CPU_PRIMITIVE_ALL_TYPE_SEQ)};
+        OF_PP_FOR_EACH_TUPLE(MAKE_NEW_FILL_ENTRY, CPU_PRIMITIVE_ALL_TYPE_SEQ CPU_PRIMITIVE_INT16_TYPE_SEQ)};
```
Comment (BBuf) [inline oneflow/core/ep/cpu/primitive/fill.cpp:69]: 由于把int16类型放到CPU_PRIMITIVE_ALL_TYPE_SEQ之后某些kernel在特定nvcc版本下会触发编译错误，暂时只在costant里面加一下

File: oneflow/core/functional/impl/comm_functor.cpp (hunk 1 above)
Comment (BBuf) [reply oneflow/core/functional/impl/comm_functor.cpp:213]: 没有，顺手改了一下这里的冗余写法

**Review by marigoold — APPROVED** [review]

File: oneflow/core/ep/cpu/primitive/type_seq.h [hunk 3]
```diff
@@ -61,7 +61,6 @@ limitations under the License.
   CPU_PRIMITIVE_BOOL_TYPE_SEQ         \
   CPU_PRIMITIVE_CHAR_TYPE_SEQ         \
   CPU_PRIMITIVE_INT8_TYPE_SEQ         \
-  CPU_PRIMITIVE_INT16_TYPE_SEQ        \
```
Comment (daquexian) [inline oneflow/core/ep/cpu/primitive/type_seq.h:64]: 这里可以添加一个注释

**Pull request merged**